tests/functional/pv/pv_services/test_automate_networkfence_workflow_stretch_cluster.py

```
@@ -0,0 +1,836 @@
+            status=constants.NODE_NOT_READY,
+            timeout=300,
+        )
+        log.info(f"Node {node_to_shutdown.name} is NotReady")
+
+        _apply_out_of_service_taint(zone_b_nodes, self._tainted_nodes)
+        _refresh_pod_state(sc_obj)
+
+        healthy_zone_b_nodes = [
+            n.name for n in zone_b_nodes if n.name != node_to_shutdown.name
+        ]
+        log.info(f"Healthy Zone-B nodes remaining: {healthy_zone_b_nodes}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)